frontend/src/modules/_shared/components/ReadoutBox.tsx

```diff
@@ -0,0 +1,140 @@
+    const maxNumItems = props.maxNumItems ?? 3;
+
+    const [flipped, setFlipped] = useState<boolean>(false);
+    const readoutRoot = useRef<HTMLDivElement>(null);
+
+    // How far away from the lower right corner the box is placed
+    // TODO: Expose as prop?
+    // ! "3" is based on the left-12/right-12 layout values.
+    const cornerDistance = parseInt(getComputedStyle(document.documentElement).fontSize) * 3;
+
+    useEffect(() => {
+        if (props.flipDisabled) return;
+
+        function maybeFlipBox(evt: MouseEvent) {
+            if (!readoutRoot.current) return;
+
+            const offsetParent = readoutRoot.current.offsetParent;
+            if (!offsetParent) return; // Not floating, I believe
+
+            const parentRect = offsetParent.getBoundingClientRect();
+            const { top, bottom, width } = readoutRoot.current.getBoundingClientRect();
+
+            // If above, or below, it's guaranteed to fit
+            if (evt.clientY < top || evt.clientY > bottom) {
+                if (flipped) setFlipped(false);
+                return;
+            }
+
+            const prefferredRight = parentRect.right - cornerDistance;
+            const prefferredLeft = parentRect.right - width - cornerDistance;
+
+            if (evt.clientX < prefferredLeft || evt.clientX > prefferredRight) {
+                if (flipped) setFlipped(false);
+            } else {
+                if (!flipped) setFlipped(true);
+            }
+        }
+
+        document.addEventListener("mousemove", maybeFlipBox);
+
+        return () => document.removeEventListener("mousemove", maybeFlipBox);
+    }, [cornerDistance, flipped, props.flipDisabled]);
+
+    if (readoutItems.length === 0) return null;
+
+    return (
+        <div
+            ref={readoutRoot}
+            className={`absolute bottom-10 right-12 z-50 w-60  flex flex-col gap-2 p-2 text-sm rounded border border-neutral-300 bg-white bg-opacity-75 backdrop-blur-sm pointer-events-none ${
+                flipped ? "left-12" : "right-12"
+            }`}
+        >
+            {readoutItems.map((item, idx) => {
+                if (idx < maxNumItems) {
+                    return (
+                        <Fragment key={idx}>
+                            <div className="flex gap-2 font-bold items-center">
+                                {!props.noLabelColor && (
+                                    <div
+                                        className="rounded-full w-3 h-3 border border-slate-500"
+                                        style={{ backgroundColor: item.color }}
+                                    />
+                                )}
+                                <span className="block">{item.label}</span>
+                            </div>
+
+                            {item.info && item.info.map((i: InfoItem, idx: number) => <InfoItem key={idx} {...i} />)}
+                        </Fragment>
+                    );
+                }
+            })}
```

Comment:
You could consider moving this into a function (e.g. `makeItems()`). This way you could break the loop early after `maxNumItems` is reached.

Reply:
Didn't do exactly as suggested, I instead shortened the item array before the render with `_.take` to avoid uneccessary loop-steps.

Also moved the label-creation code to a seperate function for better readability